### ISSUE TITLE
Newlines in process descriptions aren't allowed?

### ISSUE THREAD

**Issue by mrjbq7** [slavapestov/factor#1301]
```factor
! Bug, newlines mess things up
"\nls -l\n" utf8 [ contents ] with-process-reader
"ls -l\n" utf8 [ contents ] with-process-reader
"\nls -l" utf8 [ contents ] with-process-reader
```

**Comment by erg** [comment]
The problem is with ``simple-tokenizer``:

```factor
"\nls -l\n" tokenize
```

**Comment by mrjbq7** [comment]
Should we do this?

```diff
diff --git a/basis/simple-tokenizer/simple-tokenizer.factor b/basis/simple-tokenizer/simple-tokenizer.factor
index f7de082..8d97da7 100644
--- a/basis/simple-tokenizer/simple-tokenizer.factor
+++ b/basis/simple-tokenizer/simple-tokenizer.factor
@@ -4,7 +4,7 @@ USING: peg.ebnf strings ;
 IN: simple-tokenizer
 
 EBNF: tokenize
-space = " "
+space = " \t\n\r"
 escaped-char = "\" .:ch => [[ ch ]]
 quoted = '"' (escaped-char | [^"])*:a '"' => [[ a ]]
 unquoted = (escaped-char | [^ "])+
```

**Comment by mrjbq7** [comment]
That doesn't quite do what I meant, but you know what I mean.

**Comment by erg** [comment]
```factor
EBNF: tokenize
space = " " | "\r" | "\n" | "\t"
escaped-char = "\" .:ch => [[ ch ]]
quoted = '"' (escaped-char | [^"])*:a '"' => [[ a ]]
unquoted = (escaped-char | [^ "
] => [[ B ]] )+
argument = (quoted | unquoted) => [[ >string ]]
command = space* (argument:a space* => [[ a ]])+:c !(.) => [[ c ]]
;EBNF
```

The set negation syntax ``[^ "]`` can't handle backslashed characters, so the newline above is required, but still doesn't handle ``\r`` or ``\t`` at the end. This should be fixed but I'm having trouble finding the code for it.

**Comment by mrjbq7** [comment]
Heh, maybe we should switch from EBNF to just writing a tokenize word.

**Comment by erg** [comment]
Handwritten recursive descent parsers for life!

On Sat, Apr 18, 2015 at 5:20 PM, John Benediktsson <notifications@github.com

**Comment by mrjbq7** [comment]
I have a fix for this!

**Comment by mrjbq7** [comment]
Yay, fixed!

Had to fix ebnf parser in 56bb141257a551aff6f77037f76c173a7796ae94 and then simple-tokenizer in a30c0c0946b54f60d13f289d7c9531533c24dd83.
